test/e2e/pivoting.go

```diff
@@ -340,24 +344,41 @@ func installIronicBMO(ctx context.Context, inputGetter func() installIronicBMOIn
 	}
 }
 
+type DeploymentType string
+
+const (
+	DeploymentTypeLocal DeploymentType = "local"
+	DeploymentTypeBMO   DeploymentType = "deploy.sh"
+	DeploymentTypeIrSO  DeploymentType = "irso"
```

Comment:
There is one more deployment type, we are deploying Ironic using Kustomize in the `clusterctl upgrade tests` and not relying on BMO `deploy.sh`, as seen in the following code: https://github.com/metal3-io/cluster-api-provider-metal3/blob/6cad2fb3e1ab8274ef077c8476b75226e95e2f7b/test/e2e/upgrade_clusterctl_test.go#L236.

The primary reason for this approach was to simplify the process of installing different versions of Ironic. The plan is to use the same approach in other tests and get rid of `deploy.sh` but I haven't had the chance to work on it yet. I'll try to address this soon.

I’d like to ask how the `IrSO` manages the versioning of Ironic deployments. If it's straightforward(easy to configure), we could potentially eliminate `Kustomize` and rely solely on `IrSO`.

Reply:
Do you suggest to rename `deploy.sh` to something else or to create a new deployment type?

> I’d like to ask how the IrSO manages the versioning of Ironic deployments.

The `Ironic` CRD has a `version` field which chooses the branch of ironic-image (e.g. `27.0`) or `latest`.

Reply:
> Do you suggest to rename `deploy.sh` to something else or to create a new deployment type?

Create a new type:
`	DeploymentTypeKustomize IronicDeploymentType = "kustomize"`
Its fine not add it now, I need to do some refactoring I can add it then.



> The `Ironic` CRD has a `version` field which chooses the branch of ironic-image (e.g. `27.0`) or `latest`.

Great once this PR is merged, I will add ironic deployment IrSO in CAPM3.